src/main/java/ay2122s1_cs2103t_w16_2/btbb/logic/parser/util/ParserUtil.java

```diff
@@ -82,23 +87,105 @@ public static Email parseEmail(String email) throws ParseException {
+        requireNonNull(deadline);
+        String trimmedDeadline = deadline.trim();
+        if (!Deadline.isValidDeadline(trimmedDeadline)) {
+            throw new ParseException(Deadline.MESSAGE_CONSTRAINTS);
         }
-        return new Quantity(trimmedQuantity);
+        return new Deadline(trimmedDeadline);
+    }
+
+    /**
+     * Parses a {@code String price} into a {@code Price}.
+     * Leading and trailing whitespaces will be trimmed.
+     *
+     * @param price String input to parse.
+     * @return Price object.
+     * @throws ParseException If the given {@code price} is invalid.
+     */
+    public static Price parsePrice(String price) throws ParseException {
+        requireNonNull(price);
+        String trimmedPrice = price.trim();
+        if (!Price.isValidPrice(trimmedPrice)) {
+            throw new ParseException(Price.MESSAGE_CONSTRAINTS);
+        }
+        return new Price(trimmedPrice);
+    }
+
+    /**
+     * Parses a {@code String recipeIngredients} into a {@code RecipeIngredientList}.
+     * Leading and trailing whitespaces will be trimmed.
+     *
+     * @param recipeIngredients String input to parse.
+     * @return RecipeIngredientList object.
+     * @throws ParseException If the given {@code recipeIngredients} is invalid.
+     */
+    public static RecipeIngredientList parseRecipeIngredients(String recipeIngredients) throws ParseException {
+        requireNonNull(recipeIngredients);
+        String trimmedRecipeIngredients = recipeIngredients.trim();
+        List<Ingredient> ingredients = parseRecipeIngredientsToList(trimmedRecipeIngredients);
+        if (!RecipeIngredientList.isValidRecipeIngredientList(ingredients)) {
+            throw new ParseException(RecipeIngredientList.MESSAGE_CONSTRAINTS);
+        }
+        return new RecipeIngredientList(ingredients);
+    }
+
+    /**
+     * Parses the ingredient list to a {@code List<Ingredient>}.
+     *
+     * @param ingredientList The ingredient list to parse.
+     * @return A list of ingredients.
+     */
+    private static List<Ingredient> parseRecipeIngredientsToList(String ingredientList) throws ParseException {
+        requireNonNull(ingredientList);
+        List<Ingredient> listOfIngredients = new ArrayList<>();
+
+        String[] ingredientListArray = ingredientList.split(",");
+        for (String individualIngredient : ingredientListArray) {
+            Ingredient ingredient = parseRecipeIngredient(individualIngredient.trim());
+            listOfIngredients.add(ingredient);
+        }
+
+        return listOfIngredients;
+    }
+
+    /**
+     * Parses the individual ingredient to a {@code Ingredient}.
+     *
+     * @param individualIngredient The ingredient to parse.
+     * @return An {@code Optional<Ingredient>} object.
+     */
+    private static Ingredient parseRecipeIngredient(String individualIngredient) throws ParseException {
+        String[] individualIngredientArray = individualIngredient.split("-", 3);
```

Comment:
Is the max necessary?
```suggestion
        String[] individualIngredientArray = individualIngredient.split("-", 2);
```